docs/extending/code-overview/object-classes.rst

```diff
@@ -35,13 +35,56 @@ called and ``ExpressionList``, where each list item is either itself
 an ``ExpressionList`` or an object in a class derived from ``Atom``.
 
 The ``Atom`` class we encountered earlier when describing the nodes
-that get created intially from a parse. However there are a few other
-kinds of Atoms or fundamention objects that can appear in an
-Evaluation list. These are
+that get created intially from a parse. Those were:
 
+* ``Number``
+* ``String``
+* ``Symbol``
+* ``Filename``
+
+
+However there are a few other kinds of Atoms or fundamental objects
+that can appear in an Evaluation list.  In the evaluation process,
+other kinds of Atoms can get created. These include things like:
+
+* ``ByteArray``
 * ``CompiledCode``
+* ``Dispatch``
 * ``Image``
 
+In other words, things that might have an underlying internal representation of an object.
+
+In Mathics, the function ``AtomQ[]`` will tell you if something is an Atom, or can't be subdivided into subexpressions
+
+Some examples:
+
+.. code-block:: mathematica
+
+    (* Numbers are atoms: *)
+    >> AtomQ[1.2]
+     = True
+
+    (* So are symbolic constants: *)
+    >> AtomQ[2 + I]
+     = True
+
+    (* A 'Symbol' not bound to a value is an atom too: *)
+    >> AtomQ[x]
+     = True
+
+    (* On the other hand, expressions aren't atoms: *)
+    >> AtomQ[2 + Pi]
```

Comment:
I'm not sure, but maybe it is confusing for new users that `2 + I` is a number and `2 + Pi` is an expression, that is probably explained elsewhere, anyway...

Reply:
While I like to have clear examples, I also like examples that make the reader think a little. 

Are the the revision of examples in 73e3e0f make it more clear that `2 + I` is a complex numeric literal?

Reply:
Yes, it is much better.